extension/awsecshealthcheckextension/extension.go

```diff
@@ -0,0 +1,131 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsecshealthcheckextension
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/jaegertracing/jaeger/pkg/healthcheck"
+	"go.uber.org/zap"
+
+	"go.opencensus.io/stats/view"
+	"go.opentelemetry.io/collector/component"
+)
+
+type ECSHealthCheckExtension struct {
+	config   Config
+	logger   *zap.Logger
+	state    *healthcheck.HealthCheck
+	server   http.Server
+	interval time.Duration
+	stopCh   chan struct{}
+	exporter *ECSHealthCheckExporter
+}
+
+// initExporter function could register the customized exporter
+func (hc *ECSHealthCheckExtension) initExporter() error {
+	hc.exporter = newECSHealthCheckExporter()
+	view.RegisterExporter(hc.exporter)
+	return nil
+}
+
+func (hc *ECSHealthCheckExtension) Start(_ context.Context, host component.Host) error {
+	hc.logger.Info("Starting ECS health check extension", zap.Any("config", hc.config))
+
+	// Initialize listener
+	var (
+		ln  net.Listener
+		err error
+	)
+
+	err = hc.initExporter()
+	if err != nil {
+		return err
+	}
+
+	err = hc.config.Validate()
+	if err != nil {
+		return err
+	}
+
+	if hc.config.TCPAddr.Endpoint == "" || hc.config.TCPAddr.Endpoint == defaultEndpoint {
```

Comment:
`Endpoint` being empty should be considered error, because user explicitly overwritten it with empty.

Also why is `hc.config.TCPAddr.Endpoint == defaultEndpoint` a special case?

Reply:
Add empty endpoint in the config validation method, also delete this default endpoint check